src/datasets/packaged_modules/spark/spark.py

```diff
@@ -85,6 +87,28 @@ def _info(self):
     def _split_generators(self, dl_manager: datasets.download.download_manager.DownloadManager):
         return [datasets.SplitGenerator(name=datasets.Split.TRAIN)]
 
+    def _repartition_df_if_needed(self, max_shard_size):
+        import pyspark
+
+        def get_arrow_batch_size(it):
+            for batch in it:
+                yield pa.RecordBatch.from_pydict({"batch_bytes": [batch.nbytes]})
+
+        df_num_rows = self.df.count()
+        # Approximate the size of each row (in Arrow format) by averaging over a 100-row sample.
+        approx_bytes_per_row = (
+            self.df.limit(100)
```

Comment:
What happened if the number of rows are less than 100?

Reply:
Fixed